### PR TITLE
Add all-contributors to the repo #100

### LINKED ISSUE #100: Add all-contributors to the repo
@kentcdodds  made the coolest thing ever: [all-contributors](https://github.com/kentcdodds/all-contributors)

- Add it the readme
- Add it to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by cg-cnu**
This closes #100

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,3 +61,16 @@ API is hosted on zeit/now.
 #### Doubts?
 
 Feel free to open an issue or reach out to me [on twitter](https://twitter.com/siddharthkp).
+
+## Contributors
+
```
Comment (reznord) [inline CONTRIBUTING.md]: There no need for adding all contributors in both README & CONTRIBUTING.md

Please remove this from here.

Comment (siddharthkp) [reply CONTRIBUTING.md]: I agree, I like it in the readme 🙂

Comment (cg-cnu) [reply CONTRIBUTING.md]: Sorry, I misread the issue...
* Add it the readme
* Add it to CONTRIBUTING.md

@siddharthkp I thought you wanted to add in both the places. Will fix it.

Comment (palashmon) [reply CONTRIBUTING.md]: Also, could you please add little bit info in readme about, how a new user can add himself to this `all-contributors` in future?

Thanks!

Comment (cg-cnu) [reply CONTRIBUTING.md]: @palashmon sounds good 👍 Will add it.

Comment (siddharthkp) [reply CONTRIBUTING.md]: @cg-cnu Whoops! Sorry, that's my bad. I meant add the instructions to add yourself as a contributor, not the list of contributors

**Comment by siddharthkp** [comment]
This looks good! I'll take out some time and go through the contributions and add the other keys 😄

**Comment by cg-cnu** [comment]
You want me to add the instructions for adding new users to all-contributors ?

**Comment by siddharthkp** [comment]
> You want me to add the instructions for adding new users to all-contributors ?

Yes please!

The idea here is: If you make a new contribution, you should be able to add yourself to the contributors list yourself 😄 

This can sit in CONTRIBUTING.md

**Comment by cg-cnu** [comment]
Added the documentation for adding an user to all-contribution.
Let me know if anything needs to be worked on 😄

**Comment by palashmon** [comment]
This looks good. I think it is ready to be merged now :100:

**Comment by siddharthkp** [comment]
I see `reznord requested changes`

I'll let him approve it 😛

**Comment by palashmon** [comment]
Cool :+1:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,3 +60,19 @@ API is hosted on zeit/now.
 #### Doubts?
 
 Feel free to open an issue or reach out to me [on twitter](https://twitter.com/siddharthkp).
+
+#### Credits
+
+Once you have made some contribution. 
+
+* Pat on your back! You are oficially awesome now!!
```
Comment (reznord) [inline CONTRIBUTING.md]: typo here! it should be `officially` 

sorry 😅

Comment (siddharthkp) [reply CONTRIBUTING.md]: Nuuu, I have an issue with the copy here 😢 

This suggests that you were not awesome before!

Comment (siddharthkp) [reply CONTRIBUTING.md]: I know it's a little thing, but, let's choose to be nice when we can 😄

File: README.md [hunk 3]
```diff
@@ -148,3 +148,16 @@ MIT © [siddharthkp](https://github.com/siddharthkp)
 #### sponsor
 
 [![Sponsor](https://app.codesponsor.io/embed/LhLT2c31ydJzdLUuSR9f8mCA/siddharthkp/bundlesize.svg)](https://app.codesponsor.io/link/LhLT2c31ydJzdLUuSR9f8mCA/siddharthkp/bundlesize)
+
+## Contributors
+
+Thanks goes to these wonderful people ([emoji key](https://github.com/kentcdodds/all-contributors#emoji-key)):
```
Comment (siddharthkp) [inline README.md]: ♥️

File: CONTRIBUTING.md (hunk 2 above)
Comment (palashmon) [reply CONTRIBUTING.md]: Agreed!

Comment (cg-cnu) [reply CONTRIBUTING.md]: @siddharthkp Agree with what you said. I removed the line and did a commit. Thanks for pointing out. You guys are awesome! 😃

**Comment by reznord** [comment]
Aren't we missing the script for running `allcontributors.src`?

**Comment by kuldeepkeshwar** [comment]
I believe it is `all-contributors add <username> <contribution>`

**Comment by reznord** [comment]
yes, but if you run it like that, it should be installed globally.

**Comment by cg-cnu** [comment]
@reznord Sorry! I didn't checked that. Thanks for pointing out 😄

**Comment by reznord** [comment]
In my opinion we should shift to [`nps`](https://github.com/kentcdodds/nps).

@siddharthkp any thoughts ?

**Comment by siddharthkp** [comment]
> In my opinion we should shift to nps.

Honestly, I don't see the need right now. We will probably need it when we have > 6-7 scripts to run!

**Comment by reznord** [comment]
Makes sense

**Comment by cg-cnu** [comment]
@reznord Can you point out how do I access the locally installed cli module ?

**Comment by cg-cnu** [comment]
``npm bin all-contributors`` is this ok ? or should I add that to npm scripts ??

**Comment by reznord** [comment]
Just simply add ```addContributors: all-contributors add``` to the scripts in package.json

**Comment by cg-cnu** [comment]
@reznord Just double checking; now the final command would be 
`npm run addContributors <username> <contribution>`

**Comment by reznord** [comment]
It will be `npm run addContributors -- <username> <contribution>`

**Comment by reznord** [comment]
Instead you can just run `npm run addContributors`. It will prompt questions to the user

**Comment by cg-cnu** [comment]
I tried running both the ways. `all-contributors add` is adding the data to the rc file and readme.md but failing with the following error. 
```
undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bundlesize@0.14.1 addContributors: `all-contributors add "cg-cnu" "test"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bundlesize@0.14.1 addContributors script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2017-08-28T06_12_06_679Z-debug.log
```
Here is the [debug log](https://gist.github.com/cg-cnu/bdde6ddc2e90a6cf5c2f346edf774b94)

**Comment by cg-cnu** [comment]
Btw, am on windows 10 :(

**Comment by kuldeepkeshwar** [comment]
@cg-cnu use `./node_modules/bin/all-contributors add <username> <contribution>` if you want to run it directly from terminal

**Comment by ForsakenHarmony** [comment]
or `npx all-contributors add <username> <contribution>`

**Comment by palashmon** [comment]
Hi @cg-cnu. How to reproduce this issue on local? 

- We need to just add devDependencies `^4.4.0`, 
- then npm install, 
- then add `addContributors: "all-contributors"`  script in package.json
- then in cli run `npm run addContributors -- <username> <contribution>`

This gives the above `npm ERR! code ELIFECYCLE` error?

**Comment by cg-cnu** [comment]
Hey! Sorry guys. Am travelling, don't have access to my machine. I will try to reproduce the error and put in more details tomorrow. Will check with the main `all-contributors` repo as well.

**Comment by reznord** [comment]
> npx all-contributors add <username> <contribution>

You need to install npx for that!

**Comment by ForsakenHarmony** [comment]
heard it was installed along with npm 5 now, idk

**Comment by siddharthkp** [comment]
> heard it was installed along with npm 5 now, idk

Yep, for npm > 5.2.0
https://github.com/npm/npm/releases/tag/v5.2.0

**Comment by siddharthkp** [comment]
> Hey! Sorry guys. Am travelling, don't have access to my machine.

No worries, whenever you have time! [Remeber](https://github.com/siddharthkp/bundlesize/blob/master/maintainers.md#take-care-of-yourself)

**Comment by ForsakenHarmony** [comment]
@SaraVieira @orta danger is broken again :/

**Comment by reznord** [comment]
yup! danger broke in all PRs 😞

**Comment by SaraVieira** [comment]
@ForsakenHarmony I know ,I'm gonna fi it , we need babel

**Comment by ForsakenHarmony** [comment]
do we really? :(

**Comment by SaraVieira** [comment]
@ForsakenHarmony Yes , the version 1 is broken because of jest and we need babel , I will  keep an eye on the stuff to see when we can remove it from dev deps

**Comment by reznord** [comment]
I'm up for adding a custom check for node & npm versions if necessary instead of using `danger.js`.

I can do a quick PR if everyone is fine okay with it.

**Comment by SaraVieira** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Changes were made to package.json, but not to package-lock.json.
Perhaps you need to run `rm package-lock.json && npm install`. Make sure you’re using npm 5+.
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by reznord** [comment]
and adding it to the CI doesn't makes sense, since that check is something that needs to run on `postinstall script`

**Comment by SaraVieira** [comment]
@reznord I'm okay with danger , we just need to wait a bit to remove the babel.
Also danger is super costumizable :)

**Comment by reznord** [comment]
(just a few questions going through my mind)

-> if checking node & npm versions is something that we can handle without using babel, why do we even need to use danger.js`?

**Comment by SaraVieira** [comment]
Because the idea is to put all the errors in the danger, so if eslint breaks we use danger to show errors and it can check for a bunch of stuff. 

Also you may have npm 5 but not update the lock 

I mainly think it's a trade-off, we can do some fun stuff we danger but for now we need some more dev deps

**Comment by cg-cnu** [comment]
Hey guys, there seems to be an issue with the `all-contributors-cli` on windows. 
https://github.com/jfmengels/all-contributors-cli/issues/48
As suggested by @reznord I will just add the command ` npm run addContributors ` which seems to be the most elegant way to do it. 
Let me know if anything else needs to be updated 🙂

**Comment by cg-cnu** [comment]
Can some one confirm the command is working fine on other operating systems. I don't have access to my linux box right now 😢

**Comment by cg-cnu** [comment]
Hey @reznord! Anything else needs to be fixed in this one ?

**Comment by reznord** [comment]
I didn't check this PR whether it is still throwing that error or not. Will have to confirm it. Will do later tonight.

**Comment by cg-cnu** [comment]
@reznord Sorry for bumping this up. Just wondering if there is anything pending. If anything needs to be fixed I can work on it today and tomorrow.

**Comment by SaraVieira** [comment]
@cg-cnu can you please fix the conflicts so we can merge this ?

**Comment by cg-cnu** [comment]
@SaraVieira There is a contributors list by [open collective](https://opencollective.com/bundlesize/contributors.svg?width=890) in readme. Shall I remove that ? Adding `all-contributors` will make it redundant.

**Comment by SaraVieira** [comment]
@cg-cnu I feel really bad by saying this but maybe close this PR then? I know you worked hard here 😥

**Comment by cg-cnu** [comment]
Hey @SaraVieira Sorry for getting back so late, lost track of this one. I think we are good here. Thanks!

**Pull request closed without merging**